Meta tags

pbanaszkiewicz opened the pull request:
Add event metadata as `<meta>` tags in HTML output.

pbanaszkiewicz committed:
Add <meta> tags with workshop metadata

pbanaszkiewicz committed:
Conditionally join a list of instructors/helpers

pbanaszkiewicz commented:
Corresponding AMY PR: https://github.com/swcarpentry/amy/pull/580

wking commented on _layouts/workshop.html:
```diff
@@ -9,10 +9,26 @@
+    {% if page.instructor == "FIXME" %}
+    <meta name="instructor" content="{{page.instructor}}" />
+    {% else %}
+    <meta name="instructor" content="{{page.instructor|join:','}}" />
```
I'd rather join instructors with pipes (`|`), so we don't have to worry about folks with commas in their names (e.g. “John Smith, Jr.”).  I haven't heard of anyone with a pipe in their name, so I think we should be safe with that delimiter ;).

wking commented:
On Tue, Nov 17, 2015 at 09:39:46AM -0800, Piotr Banaszkiewicz wrote:

> Add event metatags as `<meta>` tags in HTML output.

I added a minor delimiter comment, but otherwise this looks great.
Anything that lets us decouple the data scraped by Amy from the
backing version control makes me happier ;).

pbanaszkiewicz committed:
Meta tags: change delimiter of names from "," to "|"

People can be named "John Smith, Jr.", but (probably) not
"John Smith| Jr."

pbanaszkiewicz commented:
@wking changed! By the way, adding `<meta>` tags was originally your idea.

wking commented:
On Wed, Nov 18, 2015 at 03:00:45PM -0800, Piotr Banaszkiewicz wrote:

> @wking changed!

1e19a53 looks good to me :).

> By the way, adding `<meta>` tags was originally your idea.

Ah, no wonder I like it so much :p.

pbanaszkiewicz commented:
@gvwilson @jduckles @rgaiacs anyone with merge power willing to take this forward?

The earlier we merge this PR, the shorter transition time for AMY will be. Thanks!

gvwilson commented on _layouts/workshop.html:
```diff
@@ -9,10 +9,26 @@
+    <meta name="latlng" content="{{page.latlng}}" />
+    <meta name="language" content="{{page.language}}" />
+    <meta name="eventbrite" content="{{page.eventbrite}}" />
+    {% if page.instructor == "FIXME" %}
```
This feels fragile to me - I'm 99% sure that some people will (for example) leave `instructor` blank rather than putting in `FIXME`.

pbanaszkiewicz replied:
Hi Greg,

I tested it and it appears that the `if` conditional is not needed, and all we need is `|join:'|'`.

The outputs of `{{ var | join:'|' }}` are:
- `var = ''`: `<meta name="var" content="" />`
- `var = 'FIXME'`: `<meta name="var" content="FIXME" />`
- `var = ['Piotr Banaszkiewicz']`: `<meta name="var" content="Piotr Banaszkiewicz" />`
- `var = ['Piotr Banaszkiewicz', 'Greg Wilson']`: `<meta name="var" content="Piotr Banaszkiewicz|Greg Wilson" />`

I'll remove that `if` conditionals.

pbanaszkiewicz committed:
Remove unnecessary if-conditionals from _layout/workshop.html

"join" filter only works on lists (arrays), so when `instructor`
or `helper` are strings (or empty objects) then the result will be "".

But when they are arrays of strings, then the result will be like:
"Piotr Banaszkiewicz|Greg Wilson".

pbanaszkiewicz commented:
@gvwilson updated! You should be happy now.

gvwilson commented:
:+1:
